Source/ACE/Entity/WorldObject.cs

```diff
@@ -87,6 +104,175 @@ protected WorldObject(ObjectType type, ObjectGuid guid)
+                session.Network.EnqueueSend(new GameMessageMotion(session.Player, session, MotionAutonomous.False, MovementTypes.Invalid, MotionFlags.None, MotionStance.Standing, movement1));
+
+
+                // Set Container id to 0 - you are free
+                session.Network.EnqueueSend(
+                    new GameMessageUpdateInstanceId(objectGuid, targetContainer));
+
+                var movement2 = new MovementData { ForwardCommand = 0, MovementStateFlag = MovementStateFlag.NoMotionState };
+                session.Network.EnqueueSend(new GameMessageMotion(session.Player, session, MotionAutonomous.False, MovementTypes.Invalid, MotionFlags.None, MotionStance.Standing, movement2));
+
+                // Ok, we can do the last 3 steps together.   Not sure if it is better to break this stuff our for clarity
+                // Put the darn thing in 3d space
+                // Make the thud sound
+                // Send the container update again.   I have no idea why, but that is what they did in live.
+                session.Network.EnqueueSend(
+                    new GameMessagePutObjectIn3d(session, session.Player, objectGuid),
+                    new GameMessageSound(session.Player.Guid, Sound.DropItem, (float)1.0),
+                    new GameMessageUpdateInstanceId(objectGuid, targetContainer));
+            }
+        }
+
+        /// <summary>
+        /// This is used to do the housekeeping on the server side to take an object from inventory into 3D world then tell the world
+        /// </summary>
+        public void HandlePutItemInContainer(ObjectGuid itemGuid, ObjectGuid containerGuid,  Session session)
+        {
+            lock (this.InventoryMutex)
+            {
+                // Find the item we want to pick up
+                var obj = LandblockManager.GetWorldObject(session, itemGuid);
+                if (Object.ReferenceEquals(null, obj))
+                {
+                    return;
+                    // TODO: yea, this is probably not how you do this
+                }
+                // We are picking up the  the item - let's keep track of change in burden
+                // TODO: Add check for too encumbered and send "You are too encumbered to pick that up.  Also check pack capacity
+                this.GameData.Burden += obj.GameData.Burden;
+
+                // let's move to pick up the item
+                this.PositionFlag = UpdatePositionFlag.Contact |
+                   UpdatePositionFlag.ZeroQy | UpdatePositionFlag.ZeroQx;
+                this.Position.positionX = obj.Position.positionX;
+                this.Position.positionY = obj.Position.positionY;
+                this.Position.positionZ = obj.Position.positionZ;
+                this.Position.rotationW = obj.Position.rotationW;
+                this.Position.rotationZ = obj.Position.rotationZ;
+
+                session.Network.EnqueueSend(new GameMessageUpdatePosition(this));
+
```

Comment:
remove large chunks of commented code.  If you need to keep them, feel free to make a blog page on the acemulator wiki.